### ISSUE TITLE
Regression in version 7.6.0 when base class name has more than one level of nesting

### ISSUE THREAD

**Issue by CastorAF** [mkorpela/overrides#121]
Hello,
Release 7.6.0 has broken code where the base class is qualified by several parent classes or modules:

```python
import logging.handlers

from overrides import overrides

# OK with overrides 7.5.0, throws at import with 7.6.0!
class MyHandler(logging.handlers.WatchedFileHandler):
    @overrides
    def reopenIfNeeded(self):
        pass


class Nest:
    class First:
        class Second:
            def foo(self):
                pass


# OK with overrides 7.5.0, throws at import with 7.6.0!
class ChildOfNested(Nest.First.Second):
    @overrides
    def foo(self):
        pass
```

NOTE: it works fine with one single level of qualification (e.g. `some_module.SomeClass` or `Outer.InnerClass`)

This was working fine with version 7.5.0, but is failing with version 7.6.0 with error "No super class method found" (tested with python 3.9 .7 and 3.11.3 with same error, on Linux).

The regression comes from [this change](https://github.com/mkorpela/overrides/commit/444a7a52f9b2187f12f4cfb39a8e1ac8b86ab494#diff-194fa918f067a1a88ce7abcdf6cdf93fe64db419942402c06776c1ae1c5b1479R218) (diff -213/+218 of overrides/overrides.py in changeset 444a7a5): if I comment both checks on `extends`, I get back the previous behavior. However, that change was probably done for a reason, and I don't master bytecode disassembly so I can't really propose a fix...

It's easy to work around (see below), but not trivial to figure out what's wrong! If you can have a look at it, it would be great!

Thanks in advance,
Rémi

NOTE: Possible workarounds corresponding to previous snippet:

```python
from logging.handlers import WatchedFileHandler

from overrides import overrides

# workaround: use unqualified name for the base class
class MyHandler(WatchedFileHandler):
    @overrides
    def reopenIfNeeded(self):
        pass


class Nest:
    class First:
        class Second:
            def foo(self):
                pass

# workaround: alias the nested type to an unqualified name
NestFirstSecond = Nest.First.Second
class ChildOfNested(NestFirstSecond):
    @overrides
    def foo(self):
        pass
```

**Comment by mkorpela** [comment]
Thank you for the report!

**Comment by mkorpela** [comment]
Can reproduce and added the case to main branch. Thank you again. I'll work on this asap also contributions are much appreciated.

**Comment by mkorpela** [comment]
Regression fixed in 7.7.0.

**Comment by analog-cbarber** [comment]
This problem is fixed in overrides 7.7, so we can adjust the dependency to just exclude 7.6 (or specify >=7.7)
